theano/gpuarray/opt.py

```diff
@@ -461,29 +741,32 @@ def max_inputs_to_GpuElemwise(node):
 
 @register_opt('fast_compile')
 @op_lifter([tensor.DimShuffle])
-def local_gpua_dimshuffle(node, context_name):
-    return GpuDimShuffle(node.op.input_broadcastable,
-                         node.op.new_order)
+@register_opt2([tensor.DimShuffle], 'fast_compile')
+def local_gpua_dimshuffle(op, context_name, inputs, outputs):
+    return GpuDimShuffle(op.input_broadcastable,
+                         op.new_order)
 
 
 @register_opt('fast_compile')
 @op_lifter([tensor.SpecifyShape])
-def local_gpua_specifyShape(node, context_name):
-    if isinstance(node.inputs[0].type, GpuArrayType):
+@register_opt2([tensor.SpecifyShape], 'fast_compile')
+def local_gpua_specifyShape(op, context_name, inputs, outputs):
+    if isinstance(inputs[0].type, GpuArrayType):
         return
```

Comment:
I think we should do the same here:  remove the check for GpuArrayType from the function decorated with `@register_opt2`